src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-page.component.ts

```diff
@@ -454,22 +454,33 @@ export class InstructorSessionResultPageComponent extends InstructorCommentsComp
               this.questionsModel[k].question.feedbackQuestionId,
               this.section.length === 0 ? undefined : this.section,
               this.section.length === 0 ? undefined : this.sectionType,
-          ),
         ),
-    ).pipe(finalize(() => this.isDownloadingResults = false))
+      ),
+    )
       .pipe(takeWhile(() => this.isDownloadingResults))
       .subscribe({
         next: (resp: string) => {
           out.push(resp);
           numberOfQuestionsDownloaded += 1;
           const totalNumberOfQuestions: number = Object.keys(this.questionsModel).length;
-          const progressPercentage: number = Math.round(100 * numberOfQuestionsDownloaded / totalNumberOfQuestions);
+          const progressPercentage: number = Math.round(99 * numberOfQuestionsDownloaded / totalNumberOfQuestions);
           this.progressBarService.updateProgress(progressPercentage);
         },
         complete: () => {
           if (downloadAborted) {
             return;
           }
+
+          if (this.noResponseStudents.length !== 0) {
```

Comment:
Try to reduce arrow anti-pattern by making this `if (this.noResponseStudents.length == 0) { return; }` like line 470-472.

Reply:
off the top of my head I cant really think of a good way to do this cos I still need to save the csv, close the model, etc so cant just return immediately.

Reply:
If your if block is not executed, won't your complete callback return anyways?

Reply:
I see what you mean here. It should stay as is then.